subsys/bluetooth/host/l2cap.c

```diff
@@ -70,47 +70,19 @@ NET_BUF_POOL_FIXED_DEFINE(disc_pool, 1,
 #define l2cap_lookup_ident(conn, ident) __l2cap_lookup_ident(conn, ident, false)
 #define l2cap_remove_ident(conn, ident) __l2cap_lookup_ident(conn, ident, true)
 
-struct l2cap_tx_meta_data {
-	uint16_t cid;
-	bt_conn_tx_cb_t cb;
-	void *user_data;
-};
-
-struct l2cap_tx_meta {
-	struct l2cap_tx_meta_data *data;
-};
-
-static struct l2cap_tx_meta_data l2cap_tx_meta_data_storage[CONFIG_BT_CONN_TX_MAX];
-K_FIFO_DEFINE(free_l2cap_tx_meta_data);
-
-static struct l2cap_tx_meta_data *alloc_tx_meta_data(void)
+static void *tx_cid_to_user_data(struct bt_l2cap_le_chan *chan)
 {
-	return k_fifo_get(&free_l2cap_tx_meta_data, K_NO_WAIT);
+	/* trust me bro */
+	return (void *)((uint32_t)chan->tx.cid);
 }
 
-static void free_tx_meta_data(struct l2cap_tx_meta_data *data)
-{
-	(void)memset(data, 0, sizeof(*data));
-	k_fifo_put(&free_l2cap_tx_meta_data, data);
-}
-
-#define l2cap_tx_meta_data(buf) (((struct l2cap_tx_meta *)net_buf_user_data(buf))->data)
+#define tx_cid_from_user_data(user_data) ((uint16_t)((uint32_t)user_data & 0xFFFF))
```

Comment:
```suggestion
#define tx_cid_from_user_data(user_data) ((uint16_t)(POINTER_TO_UINT(user_data) & 0xFFFF))
```

Seem superfluous to both cast to `uint16_t` and `& 0xFFFF`